### PR TITLE
Clean up train README

### PR DESCRIPTION
**Pull request by jacobfulano**

### CODE REVIEW

File: scripts/train/README.md [hunk 1]
```diff
@@ -17,7 +17,10 @@ You can read more about the benefits of using mosaicml-streaming [here](https://
 NOTE: If you only want to profile these LLMs, we recommend that you **download and prepare the `train_small` and `val_small` splits**,
 and skip the full `train` and `val` splits. You'll just need to replace `split: train` with `split: train_small`
 and `split: val` with `split: val_small` in your run YAML's dataloader config.
-You can also accomplish this in your CLI command like so: `composer train.py ... train_loader.dataset.split=train_small eval_loader.dataset.split=val_small`
+You can also accomplish this in your CLI command like so:
```
Comment (jacobfulano) [inline scripts/train/README.md:20]: We don't really introduce composer or the full correct command here. Not sure how self-contained this should all be

File: scripts/train/README.md [hunk 2]
```diff
@@ -17,8 +31,11 @@ You can read more about the benefits of using mosaicml-streaming [here](https://
 NOTE: If you only want to profile these LLMs, we recommend that you **download and prepare the `train_small` and `val_small` splits**,
 and skip the full `train` and `val` splits. You'll just need to replace `split: train` with `split: train_small`
 and `split: val` with `split: val_small` in your run YAML's dataloader config.
-You can also accomplish this in your CLI command like so: `composer train.py ... train_loader.dataset.split=train_small eval_loader.dataset.split=val_small`
-Alternatively, feel free to substitute our dataloader with one of your own in `train.py`.
+You can also accomplish this in your CLI command like so: 
+```bash
```
Comment (vchiley) [inline scripts/train/README.md:35]:
```suggestion
<!--pytest.mark.skip-->
```bash
```

**Comment by jacobfulano** [comment]
I have tried to make the README for the train folder somewhat self explanatory and consistent. I honestly think there is too much detail in there that might make it confusing for users to easily consume. In the next release I think pretraining and finetuning instructions should be in separate READMEs

File: scripts/train/README.md [hunk 3]
```diff
@@ -1,45 +1,66 @@
-# LLM Pretraining
+# LLM Pretraining and Finetuning
 
-## Installation
+This README walks through pretraining and finetuning a large language model using MosaicML's [StreamingDataset](https://github.com/mosaicml/streaming) format, [Composer](https://github.com/mosaicml/composer) trainer, and [MPT architecture](https://www.mosaicml.com/blog/mpt-7b). When used in concert on high-performance hardware such as A100 GPUs, these tools enable incredibly efficient and optimized LLM training.
```
Comment (vchiley) [inline scripts/train/README.md:3]:
```suggestion
This README walks through pretraining and finetuning a large language model using MosaicML's [StreamingDataset](https://github.com/mosaicml/streaming) format, [Composer](https://github.com/mosaicml/composer) trainer, and [MPT architecture](https://www.mosaicml.com/blog/mpt-7b). When used in concert on high-performance hardware such as A100 GPUs, these tools enable incredibly efficient and optimized LLM training.
```

File: scripts/train/README.md [hunk 4]
```diff
@@ -1,45 +1,66 @@
...
-and skip the full `train` and `val` splits. You'll just need to replace `split: train` with `split: train_small`
-and `split: val` with `split: val_small` in your run YAML's dataloader config.
-You can also accomplish this in your CLI command like so: `composer train.py ... train_loader.dataset.split=train_small eval_loader.dataset.split=val_small`
-Alternatively, feel free to substitute our dataloader with one of your own in `train.py`.
+### Converting C4 to StreamingDataset `.mds` format
 
-### Converting C4 to streaming dataset `.mds` format
 To make yourself a copy of C4, use `convert_dataset_hf.py` like so:
+
+Download the `train_small` and `val_small` splits and convert to StreamingDataset format.
```
Comment (vchiley) [inline scripts/train/README.md:41]:
```suggestion
Download the `train_small` and `val_small` splits and convert to StreamingDataset format.
```

File: scripts/train/README.md [hunk 5]
```diff
@@ -146,27 +167,31 @@ by using [Composer's logging integrations](https://docs.mosaicml.com/projects/co
...
+3. [A local or remote dataset in the StreamingDataset `.mds` format](#mdsdataset)
 
-You should probably start with ``yamls/finetune/mpt-7b_dolly_sft.yaml`
+We'll cover these in broad detail below.
 
-## Data formatting
 
-You activate finetuning via the `train_loader` and `eval_loader` fields in your configuration YAML.
-We include some reference examples inside `yamls/finetune/`.
+Example model finetuning YAML configurations can be found in [`./yamls/finetune`](./yamls/finetune). We include configurations for MPT models of various sizes, as well as T5 and Dolly.
```
Comment (vchiley) [inline scripts/train/README.md:183]:
```suggestion
Example model finetuning YAML configurations can be found in [`./yamls/finetune`](./yamls/finetune). We include configurations for MPT models of various sizes, as well as T5 and Dolly.
```

File: scripts/train/README.md [hunk 6]
```diff
@@ -146,27 +167,31 @@ by using [Composer's logging integrations](https://docs.mosaicml.com/projects/co
...
+Finetuning is enabled via the `train_loader` and `eval_loader` fields in your configuration YAML.
 
-There are 3 different types of data sources you can use for finetuning:
-(1) [the HuggingFace Hub](#1-using-a-dataset-on-the-huggingface-hub),
-(2) [a local dataset](#2-using-a-local-dataset), and
-(3) [a local or remote streaming dataset](#3-using-an-mds-formatted-streaming-dataset----locally-or-in-an-object-store).
-We'll cover these more below, but first will describe an important consideration for all 3: data formatting!
+As in the above section for pretraining, we use the same [`train.py`](train.py) script to do finetuning.
+
+* For a minimal concrete example of finetuning a GPT2 model on a locally-stored ARC-Easy dataset, see [`./finetune_example`](./finetune_example)_
```
Comment (vchiley) [inline scripts/train/README.md:188]:
```suggestion
* For a minimal concrete example of finetuning a GPT2 model on a locally-stored ARC-Easy dataset, see [`./finetune_example`](./finetune_example)
```

File: scripts/train/README.md [hunk 7]
```diff
@@ -146,27 +167,31 @@ by using [Composer's logging integrations](https://docs.mosaicml.com/projects/co
...
 
-There are 3 different types of data sources you can use for finetuning:
-(1) [the HuggingFace Hub](#1-using-a-dataset-on-the-huggingface-hub),
-(2) [a local dataset](#2-using-a-local-dataset), and
-(3) [a local or remote streaming dataset](#3-using-an-mds-formatted-streaming-dataset----locally-or-in-an-object-store).
-We'll cover these more below, but first will describe an important consideration for all 3: data formatting!
+As in the above section for pretraining, we use the same [`train.py`](train.py) script to do finetuning.
+
+* For a minimal concrete example of finetuning a GPT2 model on a locally-stored ARC-Easy dataset, see [`./finetune_example`](./finetune_example)
+
```
Comment (vchiley) [inline scripts/train/README.md:189]:
```suggestion

```

File: scripts/train/README.md [hunk 8]
```diff
@@ -146,27 +167,30 @@ by using [Composer's logging integrations](https://docs.mosaicml.com/projects/co
...
+Finetuning is enabled via the `train_loader` and `eval_loader` fields in your configuration YAML.
 
-There are 3 different types of data sources you can use for finetuning:
-(1) [the HuggingFace Hub](#1-using-a-dataset-on-the-huggingface-hub),
-(2) [a local dataset](#2-using-a-local-dataset), and
-(3) [a local or remote streaming dataset](#3-using-an-mds-formatted-streaming-dataset----locally-or-in-an-object-store).
-We'll cover these more below, but first will describe an important consideration for all 3: data formatting!
+As in the above section for pretraining, we use the same [`train.py`](train.py) script to do finetuning.
+
+* For a minimal concrete example of finetuning a GPT2 model on a locally-stored ARC-Easy dataset, see [`./finetune_example`](./finetune_example)
```
Comment (vchiley) [inline scripts/train/README.md:188]:
```suggestion
* For a minimal concrete example of finetuning a GPT2 model on a locally-stored ARC-Easy dataset, see [`./finetune_example`](./finetune_example)

```

**Review by vchiley — APPROVED** [review]

**Pull request merged**